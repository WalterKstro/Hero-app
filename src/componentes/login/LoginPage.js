import { useContext } from "react"
import AuthContext from "../../auth/AuthContext"
import useForm from "../../hooks/useForm"

const LoginPage = ({history}) => {
    const {dispatch} = useContext(AuthContext)
    const {state,handleChangeInput} = useForm({})
    
    
    const handleLogin = (e) =>{
        e.preventDefault()
        if (state.hasOwnProperty('email') && state.hasOwnProperty('pass')) {
            dispatch({
                type: 'loggin',
                payload: state
            })
        }
        /*Get the last page visited*/
        const lastPath = localStorage.getItem('lastPath') || '/' 
        history.replace(lastPath)
    }
    return (
        <div className="container">
            
            <div className="row my-5 justify-content-center">
                <div className="col-4 col-md-2 col-md-offset-5">
                    <img src={`${process.env.PUBLIC_URL}/assets/Brand.svg`} alt="Brand"/>
                </div>
            </div>
            <div className="row justify-content-center">
                <div className="col-12 col-md-6">
                    <form onSubmit={handleLogin}>
                        <div>
                            <label htmlFor="staticUsername" className="col-sm-2 col-form-label">Username</label>
                            <div className="col-sm-10">
                            <input 
                                type="text"
                                name="email" 
                                className="form-control" 
                                id="staticEmail" 
                                placeholder="your username"
                                onChange={handleChangeInput}/>
                            </div>
                            <div class="form-text">
                                Any username
                            </div>
                        </div>
                        <div className="mb-3">
                            <label htmlFor="inputPassword" className="col-sm-2 col-form-label">Password</label>
                            <div className="col-sm-10">
                            <input 
                                type="password" 
                                name="pass"
                                className="form-control" 
                                id="inputPassword" 
                                placeholder="********"
                                onChange={handleChangeInput}
                                />
                                <div class="form-text">
                                    Any password
                                </div>
                            </div>
                        </div>
                        <div>
                        <button 
                            type="submit" 
                            className="btn btn-primary"
                            >Log In</button>
                        </div>
                    </form>
                </div>
            </div>
            
        </div>
    )
}

export default LoginPage
