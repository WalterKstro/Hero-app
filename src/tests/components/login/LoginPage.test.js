import { mount } from "enzyme"
import AuthContext from "../../../auth/AuthContext"
import LoginPage from "../../../componentes/login/LoginPage"


describe('Testing in <LoginPage/>', () => {

    let history = undefined
    let rendered = undefined

    const user = {
        email : 'Walter Francisco',
        pass :'123'
    }
    const dispatch = jest.fn()


    beforeEach(() => {
        history = {
            replace : jest.fn()
        }

        rendered = mount(
            <AuthContext.Provider value={{user, dispatch}}>
                <LoginPage  history = {history}/>
            </AuthContext.Provider>
        )
    })

    test('should display correctly', () => {
        expect(rendered).toMatchSnapshot()
    })

    test('should execute the function replace', () => {
        rendered.find('form').simulate('submit')
        
        expect(history.replace).toHaveBeenCalledWith('/')
        expect(history.replace).toHaveBeenCalledTimes(1)
        
    })

    test('should get the last URL visited of the localStorage', () => {
        

        localStorage.setItem('lastPath','/dc')
        rendered.find('form').simulate('submit')
        expect(history.replace).toHaveBeenCalledWith('/dc')

    })
    
    
    
})
