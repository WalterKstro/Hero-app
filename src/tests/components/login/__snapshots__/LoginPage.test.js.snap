// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <LoginPage/> should display correctly 1`] = `
<div
  className="container"
>
  <div
    className="row my-5 justify-content-center"
  >
    <div
      className="col-4 col-md-2 col-md-offset-5"
    >
      <img
        alt="Brand"
        src="/assets/Brand.svg"
      />
    </div>
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-12 col-md-6"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            className="col-sm-2 col-form-label"
            htmlFor="staticUsername"
          >
            Username
          </label>
          <div
            className="col-sm-10"
          >
            <input
              className="form-control"
              id="staticEmail"
              name="email"
              onChange={[Function]}
              placeholder="your username"
              type="text"
            />
          </div>
          <div
            className="form-text"
          >
            Any username
          </div>
        </div>
        <div
          className="mb-3"
        >
          <label
            className="col-sm-2 col-form-label"
            htmlFor="inputPassword"
          >
            Password
          </label>
          <div
            className="col-sm-10"
          >
            <input
              className="form-control"
              id="inputPassword"
              name="pass"
              onChange={[Function]}
              placeholder="********"
              type="password"
            />
            <div
              className="form-text"
            >
              Any password
            </div>
          </div>
        </div>
        <div>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Log In
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
